src/test/test-flow.c

```diff
@@ -1151,12 +1152,14 @@ named_options_init_from_strv(void)
 {
     struct sol_flow_node_named_options named_opts;
     struct sol_flow_node_named_options_member *m;
+    struct sol_flow_node_type **node_type;
     int r;
 
+    node_type = sol_symbol_get("flow", "timer", SOL_FLOW_NODE_TYPE_TIMER);
     {
         const char *strv[] = { "interval=1000", NULL };
 
-        r = sol_flow_node_named_options_init_from_strv(&named_opts, SOL_FLOW_NODE_TYPE_TIMER, strv);
+        r = sol_flow_node_named_options_init_from_strv(&named_opts, *node_type, strv);
```

Comment:
On Tue, 22 Sep 2015 12:10:05 -0700, Gustavo Sverzut Barbieri wrote:
> >      {
> >          const char *strv[] = { "interval=1000", NULL };
> >  
> > -        r = sol_flow_node_named_options_init_from_strv(&named_opts, SOL_FLOW_NODE_TYPE_TIMER, strv);
> > +        r = sol_flow_node_named_options_init_from_strv(&named_opts, *node_type, strv);
> 
> crashes if `node_type == NULL`.
> 
> What if we change to:
> ```c
> struct sol_flow_node_type *node_type; /* pointer, not pointer to pointer */
> if (sol_symbol_get("flow", "timer", SOL_FLOW_NODE_TYPE_TIMER, &node_type) < 0)
>    goto error;
> 
>  r = sol_flow_node_named_options_init_from_strv(&named_opts, node_type, strv);
> ```

Those I just ignored, if the test crashes, someone fucked up anyway.

And passing the variable as a parameter, when modules are entirely
disabled would end up breaking, as we can't do:

&node_type = SOL_FLOW_NODE_TYPE_TIMER

There's the option of adding the & in the macro, but I dislike that.